tests/integration/components/sidenotes-wrapper-test.js

```diff
@@ -204,4 +205,102 @@ module('Integration | Component | sidenotes-wrapper', function (hooks) {
       top: `${this.items[1].y + elementHeight}px`,
     });
   });
+
+  test('it triggers placement on items change', async function (assert) {
+    this.set('onSidenotesMovedCalled', false);
+    this.set('onSidenotesMoved', () => {
+      this.set('onSidenotesMovedCalled', true);
+    });
+
+    this.set('style', htmlSafe(`height: 100px;`));
+
+    await render(hbs`
+      <SidenotesWrapper
+        @items={{this.items}}
+        @onSidenotesMoved={{this.onSidenotesMoved}}
+        as |Sidenote item|
+      >
+        <Sidenote
+          data-sidenote-id={{item.id}}
+          style={{this.style}}
+          @id={{item.id}}
+          @offsetY={{item.y}}
+          @data={{item.data}}
+        />
+      </SidenotesWrapper>
+    `);
+
+    const element = this.items[2];
+
+    await waitUntil(() => this.onSidenotesMovedCalled);
+    this.set('onSidenotesMovedCalled', false);
+    assert.dom('[data-sidenote-id="3"]').doesNotHaveStyle({
+      top: `${element.y}px`,
+    });
+
+    this.set(
+      'items',
+      this.items.filter(({ id }) => id !== 2)
+    );
+
+    await waitUntil(() => this.onSidenotesMovedCalled);
+    this.set('onSidenotesMovedCalled', false);
+    assert.dom('[data-sidenote-id="3"]').hasStyle({
+      top: `${element.y}px`,
+    });
+  });
+
+  test('it replaces sidenote with same id', async function (assert) {
+    await render(hbs`
+      <SidenotesWrapper
+        @items={{this.items}}
+        as |Sidenote item|
+      >
+        <Sidenote
+          data-sidenote-id={{item.id}}
+          @id={{item.id}}
+          @offsetY={{item.y}}
+          @data={{item.data}}
+        >
+          Dummy text
+        </Sidenote>
+      </SidenotesWrapper>
+    `);
+
+    const { id } = this.items[0];
+
+    assert.dom(`[data-sidenote-id="${id}"]`).isVisible({ count: 1 });
+
+    this.set('items', [
+      ...this.items.filter(({ id: itemId }) => itemId !== id),
+      { id, y: 100, data: null },
+    ]);
```

Comment:
```suggestion
    this.set('items', [
      ...this.items,
      { id, y: 100, data: null },
    ]);
```

the whole filtering thing is not necessary

Reply:
This is necessary because we only avoid duplicate in `notes` to prevent having 2 notes when registering two times the same note (in terms of id). We don't restrict @items on display so if there are 2 items with same id, both will be displayed but only the last will be handled for placement.

Should we use a getter to get `@items` with duplicates id removed to prevent that too ? :thinking:

Reply:
:thinking: Ok I'm not sure to understand what you are testing here :thinking: